signer/src/stacks/api.rs

```diff
@@ -250,7 +249,10 @@ impl TryFrom<AccountEntryResponse> for AccountInfo {
 pub struct StacksClient {
     /// The base URL (with the port) that will be used when making requests
     /// for to a Stacks node.
-    pub node_endpoint: url::Url,
+    pub node_endpoints: Vec<url::Url>,
+    /// The current index into the endpoints list
+    /// Increment this on network failure
+    pub endpoint_index: usize,
```

Comment:
I think we should make these fields private, since there are certain invariants that the struct needs to uphold. The `node_endpoints` must be non-empty and the `endpoint_index` value is always less than the `node_endpoints`' length.

---

As an aside, you could have a `node_endpoint` field that is still a `url::Url` and a `endpoints` field that is a `std::iter::Cycle<std::vec::IntoIter<url::Url>>`. Getting the next endpoint is then just 
```rust
// You can unwrap because an invariant is the `Vec` that created
// the `IntoIter` was nonempty.
self.node_endpoint = self.endpoints.next().unwrap();
```
Eh, actually, that might be overkill. Getting the endpoints the way that you have here is fine and more obvious if since you don't need to know about `std::iter::Cycle` type.

Reply:
endpoint fields are now private